### PR TITLE
Adding ss module 25th latest

### PR DESCRIPTION
**Pull request by ShruthiNiranjan**

### CODE REVIEW

File: src/test/java/SalesforceModules/CasesUtil.java [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+
+    }
+
+    /**
+     * Enter phone number field for new account.
+     *
+     * @param logStep the log
+     * @param PhoneNumber1 the phoneNumber1
+     */
+    public static void enterPhoneNumber1(String PhoneNumber1, String logStep){
```
Comment (snaphray) [inline src/test/java/SalesforceModules/CasesUtil.java]: Please update the argument name to phoneNumber

File: src/test/java/SalesforceModules/CasesUtil.java [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+    }
+
+    /**
+     * Enter account name field for new account.
+     *
+     * @param logStep the log
+     * @param AccountName1 the account name
+     * @param PhoneNumber1 the PhoneNumber1
+     */
+    public static void newCaseByCreatingNewAccount(String AccountName1, String PhoneNumber1,String origin,String logStep) throws InterruptedException {
```
Comment (snaphray) [inline src/test/java/SalesforceModules/CasesUtil.java]: Change arguments name from AccountName1 to accountName and PhoneNumber1 to phoneNumber, it should start from small letter

File: src/test/java/SalesforceModules/ContactUtil.java [hunk 3]
```diff
@@ -156,4 +157,47 @@ public static void createContact(String salutation, String lastname, String emai
...
+        JavascriptExecutor executor = (JavascriptExecutor)KeywordUtil.getDriver();
+        executor.executeScript("arguments[0].click();", element);
+    }
+
+    /**
+     * click new button for creating new contact for service support.
+     *
+     * @param logStep the log
+     */
+    public static void NewButtonContact_ss(String logStep) throws InterruptedException {
```
Comment (snaphray) [inline src/test/java/SalesforceModules/ContactUtil.java]: Method Name should start from small letter

File: src/test/java/SalesforceModules/EditandDeleteUtil.java [hunk 4]
```diff
@@ -112,4 +116,229 @@ public static void deleteAccount(String logStep) throws InterruptedException {
...
+
+    /**
+     * click drop down in contacts section.
+     *
+     * @param logStep the log
+     */
+    public static void clickDropDownButton_ss(String logStep){
+        KeywordUtil.waitForVisible(ServiceSupportObject.dropdown_ss);
+        KeywordUtil.clickJS(ServiceSupportObject.dropdown_ss,logStep);
+       // KeywordUtil.click(ServiceSupportObject.dropdown_ss,logStep);
```
Comment (snaphray) [inline src/test/java/SalesforceModules/EditandDeleteUtil.java]: remove the commented line

File: src/test/java/pageobjects/SalesforceObjects/ServiceSupportObject.java [hunk 5]
```diff
@@ -14,4 +14,74 @@ public class ServiceSupportObject {
     public static By dropDown = By.xpath("//th[span[a[text()='TXTest']]]/following-sibling::td//a[@role='button']");
     public static By deleteButton = By.xpath("//a[@data-target-selection-name='sfdc:StandardButton.Account.Delete']");
     public static By clickdeleteButton = By.xpath("//button[@title='Delete']");
+
+    //Contact xpaths
+    public static By newContactButton_ss = By.xpath("//button[@class='slds-button slds-button_neutral middleButton actionListButton' and text()='New']");
+    //public static By dropdown_ss = By.xpath("(//button[@class='slds-button slds-button_icon-border-filled fix-slds-button_icon-border-filled slds-button_last'])[9]");
```
Comment (snaphray) [inline src/test/java/pageobjects/SalesforceObjects/ServiceSupportObject.java]: remove the commented line

File: src/test/java/step_definitions/SalesForceServiceSupportDemo.java [hunk 6]
```diff
@@ -28,4 +42,81 @@ public void delete_account_details() throws Exception{
...
+    }
+
+    @When("^Remove All Filters Contact Details for service support")
+    public void removeAll_filter_contact_details() throws Exception{
+        EditandDeleteUtil.removeFilter_cnt();
+
+    }
+
+    @When("^Delete Contact Details for service support")
+    public void Delete_contact_details() throws Exception {
```
Comment (snaphray) [inline src/test/java/step_definitions/SalesForceServiceSupportDemo.java]: function name should start from small letter

File: src/test/java/step_definitions/SalesForceServiceSupportDemo.java [hunk 7]
```diff
@@ -28,4 +42,81 @@ public void delete_account_details() throws Exception{
...
+        CasesUtil.deleteCase("Deleted the contact");
+    }
+
+    @When("^Search Case for service support")
+    public void search_case() throws Exception{
+        CasesUtil.searchCase("search the case from the list");
+    }
+
+    @When("Create New Case by Creating an New Account")
+    public void createNewCaseByNewAccount() throws Exception {
```
Comment (snaphray) [inline src/test/java/step_definitions/SalesForceServiceSupportDemo.java]: Follow the same pattern throughout the file either include underscore in the function names or don't use them at all

File: src/test/resources/features/SalesForceServiceSupport.feature [hunk 8]
```diff
@@ -1,14 +1,27 @@
...
-    Then create new account
-    Then Edit Account Details
-    Then Delete Account
+#    Then create account for service support
+#    Then Edit Account Details
+#    Then create contact for service support
+#    Then Edit Contact Details for service support
+#    Then Filter Contact Details for service support
+#    Then Remove All Filters Contact Details for service support
+#    Then Click on CasesTab for service support
```
Comment (snaphray) [inline src/test/resources/features/SalesForceServiceSupport.feature]: remove the commented line if not required

**Review by snaphray — CHANGES_REQUESTED** [review]
Please resolve review comments

File: src/test/resources/features/SalesForceServiceSupport.feature (hunk 8 above)
Comment (ShruthiNiranjan) [reply src/test/resources/features/SalesForceServiceSupport.feature]: ok

File: src/test/java/step_definitions/SalesForceServiceSupportDemo.java (hunk 7 above)
Comment (ShruthiNiranjan) [reply src/test/java/step_definitions/SalesForceServiceSupportDemo.java]: done

**Review by snaphray — APPROVED** [review]
Code looks fine

**Pull request merged**